test/hotspot/jtreg/serviceability/dcmd/gc/HeapDumpParallelTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2023, Alibaba Group Holding Limited. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
+import jdk.test.lib.Asserts;
+import jdk.test.lib.JDKToolLauncher;
+import jdk.test.lib.apps.LingeredApp;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+
+import jdk.test.lib.hprof.HprofParser;
+
+/**
+ * @test
+ * @bug 8306441
+ * @summary Verify the integrity of generated heap dump and capability of parallel dump
+ * @library /test/lib
+ * @run driver HeapDumpParallelTest
+ */
+
+class HeapDumpParallel extends LingeredApp {
+    public static void main(String[] args) {
+        System.out.println("Hello world");
+        LingeredApp.main(args);
+    }
+}
+
+public class HeapDumpParallelTest {
+    static HeapDumpParallel theApp;
+
+    private static void checkAndVerify(OutputAnalyzer out, LingeredApp app, File heapDumpFile, boolean expectSerial) {
+        out.shouldHaveExitValue(0);
+        out.shouldContain("Heap dump file created");
+        if (!expectSerial && Runtime.getRuntime().availableProcessors() > 1) {
+            Asserts.assertTrue(app.getProcessStdout().contains("Dump heap objects in parallel"));
```

Comment:
I think we just need these other asserts converted to shouldContain, and the assertFalse changed to shouldNotContain.
This might be a annoying, but if you get the test failing and don't get the actual output, that's worse. 8-)
Maybe we can aim to integrate on Monday.

Reply:
getProcessStdout is actually a string instead of OutputAnalyzer, its name is somewhat deceptive